### PR TITLE
Voting system

### PR DESCRIPTION
**Pull request by anyulled**
solves #9

### CODE REVIEW

File: db/migration/V5__initial_data.sql
Comment (alexdeveng) [inline db/migration/V5__initial_data.sql:1]: In spring boot you have the idea of schema.sql or data.sql just in case you want to take that into consideration with the naming

File: README.md
Comment (alexdeveng) [inline README.md:11]: Ping me about this line :)

File: documentation/votes.http
Comment (alexdeveng) [inline documentation/votes.http:2]: We need to agree with this because I used curl for the demos. Or maybe we can use openapi

File: src/test/java/com/acme/conferencesystem/ContainerConfig.java
Comment (alexdeveng) [inline src/test/java/com/acme/conferencesystem/ContainerConfig.java:20]: TODO

File: src/test/java/com/acme/conferencesystem/voting/http/VoteControllerIntegrationTest.java
Comment (alexdeveng) [inline src/test/java/com/acme/conferencesystem/voting/http/VoteControllerIntegrationTest.java:49]: You'll have an error, newUserId is an Object and you need a UUID, otherwise, Instancio will not be able to create the object and it'll return null.

```java
  // Submit user
        String newUserId = given(requestSpecification)
                .contentType(ContentType.JSON)
                .body(user)
                .when()
                .post("/users")
                .then()
                .statusCode(201)
                .extract()
                .path("id");

        // 2. create a proposal
        Proposal proposal = Instancio.of(Proposal.class)
                .ignore(field(Proposal::id))
                .set(field(Proposal::speakerId), UUID.fromString(newUserId))
                .create();
```

File: db/migration/V5__initial_data.sql
Comment (anyulled) [reply db/migration/V5__initial_data.sql]: done

File: documentation/votes.http
Comment (anyulled) [reply documentation/votes.http:2]: Let's use markdown and HTTP requests both

File: src/test/java/com/acme/conferencesystem/ContainerConfig.java
Comment (anyulled) [reply src/test/java/com/acme/conferencesystem/ContainerConfig.java]: removed

File: src/test/java/com/acme/conferencesystem/voting/http/VoteControllerIntegrationTest.java
Comment (anyulled) [reply src/test/java/com/acme/conferencesystem/voting/http/VoteControllerIntegrationTest.java]: Thanks

File: src/main/java/com/acme/conferencesystem/cfp_proposals/business/ProposalService.java
Comment (alexdeveng) [inline src/main/java/com/acme/conferencesystem/cfp_proposals/business/ProposalService.java:80]: This should be updatedate not creation date

Comment (alexdeveng) [reply src/main/java/com/acme/conferencesystem/cfp_proposals/business/ProposalService.java:80]: if we want to be purist we will need a new column

Comment (alexdeveng) [reply src/main/java/com/acme/conferencesystem/cfp_proposals/business/ProposalService.java:80]: but well let's merge and we can do that in another MR

also  we can do something like this


```java
        repository.findById(id).map(proposalEntity -> new ProposalEntity(
                proposalEntity.id(),
                proposalEntity.title(),
                proposalEntity.description(),
                proposalEntity.speakerId(),
                LocalDateTime.now(),
                ProposalStatus.ACCEPTED
        )).ifPresent(repository::save);
```

Comment (alexdeveng) [reply src/main/java/com/acme/conferencesystem/cfp_proposals/business/ProposalService.java:80]: and also we can do a factory method like proposalEntity.withStatus(ACCEPTED);

File: src/main/java/com/acme/conferencesystem/cfp_proposals/http/ProposalController.java
Comment (alexdeveng) [inline src/main/java/com/acme/conferencesystem/cfp_proposals/http/ProposalController.java:62]: If Patch doesn't return content, The response status for no content should be 204 not 200

File: src/main/java/com/acme/conferencesystem/cfp_proposals/http/ProposalController.java
Comment (alexdeveng) [inline src/main/java/com/acme/conferencesystem/cfp_proposals/http/ProposalController.java:56]: Same here

File: src/main/java/com/acme/conferencesystem/cfp_proposals/http/ProposalController.java
Comment (alexdeveng) [reply src/main/java/com/acme/conferencesystem/cfp_proposals/http/ProposalController.java:62]: But maybe will be more practical to return the resource itself, in this case OK http status will be fine.

Comment (alexdeveng) [reply src/main/java/com/acme/conferencesystem/cfp_proposals/http/ProposalController.java:62]: https://google.aip.dev/121

Comment (alexdeveng) [reply src/main/java/com/acme/conferencesystem/cfp_proposals/http/ProposalController.java:62]: but let's merge ;)

File: src/main/java/com/acme/conferencesystem/voting/business/VoteService.java
Comment (alexdeveng) [inline src/main/java/com/acme/conferencesystem/voting/business/VoteService.java:6]: not sure if we can use import jakarta.validation.constraints.NotNull; instead.

File: src/main/java/com/acme/conferencesystem/voting/business/VoteService.java
Comment (alexdeveng) [inline src/main/java/com/acme/conferencesystem/voting/business/VoteService.java:18]: I think that Spring 4.3 is not necessary to explicitly use @Autowired

https://spring.io/blog/2016/03/04/core-container-refinements-in-spring-framework-4-3

File: src/test/java/com/acme/conferencesystem/ArchitectureTests.java
Comment (alexdeveng) [inline src/test/java/com/acme/conferencesystem/ArchitectureTests.java:27]: comment or delete this xD

new Documenter(modules).writeDocumentation(); this does everything and the commented code is what it does

Comment (alexdeveng) [reply src/test/java/com/acme/conferencesystem/ArchitectureTests.java:27]: Have a look of the implementation of writeDocumentation

File: src/test/java/com/acme/conferencesystem/voting/persistence/ContainerVoteTestConfig.java
Comment (alexdeveng) [inline src/test/java/com/acme/conferencesystem/voting/persistence/ContainerVoteTestConfig.java:22]: Why do we need this?

File: src/test/java/com/acme/conferencesystem/voting/persistence/VoteRepositoryTest.java
Comment (alexdeveng) [inline src/test/java/com/acme/conferencesystem/voting/persistence/VoteRepositoryTest.java:46]: Not very happy using the repositories of the other modules in the voting repository test. But well we can discuss about it in the future ;)

Comment (alexdeveng) [reply src/test/java/com/acme/conferencesystem/voting/persistence/VoteRepositoryTest.java:46]: I'm thinking about the future migration into a microservice

File: src/main/java/com/acme/conferencesystem/cfp_proposals/business/ProposalService.java
Comment (anyulled) [reply src/main/java/com/acme/conferencesystem/cfp_proposals/business/ProposalService.java:80]: yes, I was planning to optimize the code after finishing all the pending work

File: src/main/java/com/acme/conferencesystem/cfp_proposals/http/ProposalController.java
Comment (anyulled) [reply src/main/java/com/acme/conferencesystem/cfp_proposals/http/ProposalController.java:62]: Yes, I was also considering return the resource itself. I'll do the changes in a separate PR, I just wanted to finish soon and unblock you

File: src/main/java/com/acme/conferencesystem/voting/business/VoteService.java
Comment (anyulled) [reply src/main/java/com/acme/conferencesystem/voting/business/VoteService.java:18]: Thanks, I only added it here because an unrelated error appeared.

File: src/test/java/com/acme/conferencesystem/voting/persistence/ContainerVoteTestConfig.java
Comment (anyulled) [reply src/test/java/com/acme/conferencesystem/voting/persistence/ContainerVoteTestConfig.java:22]: We don't

**Comment by anyulled** [comment]
@alexdeveng I need your approval for automatic merging of the pull request

**Review by alexdeveng — APPROVED** [review]

**Pull request merged**
